File: basecommon/src/main/java/qzl/com/basecommon/ui/HeadControlPanel.java
package qzl.com.basecommon.ui;

import android.content.Context;
import android.graphics.Color;
import android.os.Build;
import android.util.AttributeSet;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.RelativeLayout;
import android.widget.TextView;
import qzl.com.basecommon.R;

/**
 * Created by 魏荣 on 2015/05/15.
 */
public class HeadControlPanel extends RelativeLayout {
    private Context mContext;
    private TextView mMidleTitle;
    private ImageView mLeftImage;
    private TextView mRightText;
    private ImageView mRightImg;
    private TextView mLeftText;
    private ImageView mLeftImg;
    private LinearLayout headLayoutBack;
    private View head_view;
    private static final float middle_title_size = 20f;
    private static final float title_size = 18f;
    private static final int default_background_color = Color.rgb(99, 220, 215);

    public HeadControlPanel(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    protected void onFinishInflate() {
        super.onFinishInflate();
        mMidleTitle = (TextView) findViewById(R.id.midle_title);
        mRightText = (TextView) findViewById(R.id.right_btn_text);
        mLeftText = (TextView) findViewById(R.id.left_btn_text);
        mLeftImage = (ImageView) findViewById(R.id.menu_btn_img);
        mRightImg = (ImageView) findViewById(R.id.right_btn_img);
        mLeftImg = (ImageView) findViewById(R.id.left_btn_img);
        headLayoutBack = (LinearLayout) findViewById(R.id.head_layout_back);
        head_view = (View) findViewById(R.id.head_view);
        if (!isInEditMode()){
            if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
                if (head_view != null) {
                    head_view.setVisibility(VISIBLE);
                }
            }
        }
        //setBackgroundResource(R.color.cell_ordinary_m);
    }

    public void initHeadPanel() {
        if (mMidleTitle != null) {
            setMiddleTitle("首页");
        }
    }

    public void setMiddleTitle(String s) {
        if ("河长制移动平台".equals(s)) {
            s = "河长制移动平台";
        }
        if ("河长制移动平台".equals(s)) {
            mMidleTitle.setTextSize(middle_title_size);
            mMidleTitle.setText(s);
        } else {
            mMidleTitle.setTextSize(title_size);
            if(s.length() == 2){
               s = s.charAt(0) + " " + s.charAt(1);
            }
            mMidleTitle.setText(s);
        }
    }

    public ImageView getmLeftImage() {
        return mLeftImage;
    }

    public TextView getmRightText() {
        return mRightText;
    }
    public ImageView getmRightImg() {
        return mRightImg;
    }

    public TextView getLeftText() {
        return mLeftText;
    }

    public ImageView getLeftImg() {
        return mLeftImg;
    }

    public LinearLayout getHeadLayoutBack() {
        return headLayoutBack;
    }
}

File: basecommon/src/main/java/qzl/com/basecommon/ui/kotlin/HeadControlPanel.kt
package qzl.com.basecommon.ui.kotlin

import android.content.Context
import android.graphics.Color
import android.os.Build
import android.util.AttributeSet
import android.view.View
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.RelativeLayout
import android.widget.TextView
import kotlinx.android.synthetic.main.head_control_panel.view.*
import qzl.com.basecommon.R

/**
 * @author 强周亮(Qzl)
 * @desc 头部导航
 * @email 2538096489@qq.com
 * @time 2019-05-28 19:22
 * @class HeadControlPanel
 * @package qzl.com.basecommon.ui
 */
class HeadControlPanel(context: Context, attrs: AttributeSet) : RelativeLayout(context, attrs) {
    var leftText: TextView? = null
        private set
    var leftImg: ImageView? = null
        private set
    var headLayoutBack: LinearLayout? = null
        private set

    override fun onFinishInflate() {
        super.onFinishInflate()
        leftText = findViewById<View>(R.id.left_btn_text) as TextView
        leftImg = findViewById<View>(R.id.left_btn_img) as ImageView
        headLayoutBack = findViewById<View>(R.id.head_layout_back) as LinearLayout
        if (!isInEditMode && Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            head_view.visibility = View.VISIBLE
        }
    }

    fun initHeadPanel() {
        setMiddleTitle("首页")
    }

    fun setMiddleTitle(s: String) {
        var s = s
        midle_title.textSize = title_size
        if (s.length == 2) {
            s = s[0] + " " + s[1]
        }
        midle_title.text = s
    }

    fun getmLeftImage(): ImageView? {
        return menu_btn_img
    }

    fun getmRightText(): TextView? {
        return right_btn_text
    }

    fun getmRightImg(): ImageView? {
        return right_btn_img
    }

    companion object {
        private val middle_title_size = 20f
        private val title_size = 18f
        private val default_background_color = Color.rgb(99, 220, 215)
    }
}
